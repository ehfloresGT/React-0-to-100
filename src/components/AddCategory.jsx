import {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const onInputValueChange = ({target}) => {
    setInputValue(target.value);
  };

  const onSubmitForm = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmitForm} aria-label='form'>
      <input
        type="text"
        placeholder="Buscar gifs ..."
        value={inputValue}
        onChange={onInputValueChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
