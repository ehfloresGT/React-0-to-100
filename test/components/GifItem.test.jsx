import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Testing ... <GifItem />', () => {

  const image = {
    id: 'VXJWhaO7afRe',
    title: 'One Punch Man GIF',
    url: 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=557e256fq6eblsak90npnnqmv5i04n5hi9emusmg03q5dym5&ep=v1_gifs_search&rid=giphy.gif&ct=g',
  };
  
  test('Generar y validar con el snapshot ...', () => {
    const {container} = render(<GifItem title={image.title} url={image.url} />);
    expect(container).toMatchSnapshot();
  });

  test('Mostrar la imagen con el URL y el ALT indicado.', () => {
    render(<GifItem title={image.title} url={image.url} />);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });

  test('Mostrar el titulo en el componente.', () => {
    render(<GifItem title={image.title} url={image.url} />);
    expect(screen.getByText(image.title)).toBeTruthy();
  });

});
