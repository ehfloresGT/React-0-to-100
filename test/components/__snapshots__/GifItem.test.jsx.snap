// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ... <GifItem /> Generar y validar con el snapshot ... 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=557e256fq6eblsak90npnnqmv5i04n5hi9emusmg03q5dym5&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
